realm/realm-library/src/main/java/io/realm/RealmAny.java

```diff
@@ -0,0 +1,493 @@
+/*
+ * Copyright 2020 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm;
+
+import org.bson.types.Decimal128;
+import org.bson.types.ObjectId;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Date;
+import java.util.UUID;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+
+/**
+ * {@link io.realm.RealmAny} is used to represent a polymorphic Realm value.
+ * <p>
+ * At any particular moment an instance of this class stores a
+ * definite value of a definite type. If, for instance, that is an
+ * double value, you may call asDouble() to extract that value. You
+ * may call getType() to discover what type of value is currently
+ * stored. Calling asDouble() on an instance that does not store an
+ * double would raise a {@link java.lang.ClassCastException}.
+ * <p>
+ * RealmAny behaves like a value type on all the supported types except on
+ * Realm objects. It means that Realm will not persist any change to the
+ * RealmAny value except when the type is Realm object. Because RealmAny
```

Comment:
I would extend the wording for Realm object....perhaps something like this 

```
..when the type is Realm object. When a RealmAny holds a Realm object, it just holds the reference to it, not a copy of the object. So modifications to the Realm object is reflected in the RealmAny value, including if the object is deleted.


````